test/dynamic_shape_model_test2.py

```diff
@@ -0,0 +1,56 @@
+from sklearn.datasets import make_blobs
+import torch
+import torch_xla.core.xla_model as xm
+
+pd = torch._C._EnablePythonDispatcher()
+dev = xm.xla_device()
+
+# SIMPLE NN MODEL
+class Feedforward(torch.nn.Module):
+        def __init__(self, input_size, hidden_size):
+            super(Feedforward, self).__init__()
+            self.input_size = input_size
+            self.hidden_size  = hidden_size
+            self.fc1 = torch.nn.Linear(self.input_size, self.hidden_size)
+            self.relu = torch.nn.ReLU()
+            self.fc2 = torch.nn.Linear(self.hidden_size, 1)
+            self.sigmoid = torch.nn.Sigmoid()
+        def forward(self, x):
+            hidden = self.fc1(x)
+            relu = self.relu(hidden)
+            output = self.fc2(relu)
+            output = self.sigmoid(output)
+            return output
+
+# CREATE FAKE TRAIN DATA
+num_features = 2
+x_train, y_train = make_blobs(n_samples=40, n_features=num_features, cluster_std=1.5, shuffle=True)
+x_train = torch.Tensor(x_train)
+y_train = torch.Tensor(y_train)
+
+# CREATE FAKE TEST DATA
+num_test_samples = 5
+x_test = torch.ones_like(torch.empty(num_test_samples, num_features))
+x_test[0][0] = 0
+y_test = torch.ones_like(torch.empty(num_test_samples*2-1))
```

Comment:
Shouldn't it be:

```
y_test = torch.ones_like(torch.empty(num_test_samples*2))
y_test[0] = 0
y_test = y_test.to(dev)
y_test = torch.nonzero(y_test)
```

The above sets `y_test` upper bound to `20` and it's dynamic bound to `19` which is consistent with the characteristics of `x_test`. Wdyt @vanbasten23?

Reply:
> did you verify this implementation works? since `y_test` is not dynamic, I would imaging line 51 fails. No?

Yes it would fail at line 51.

Reply:
> Shouldn't it be:
> 
> ```
> y_test = torch.ones_like(torch.empty(num_test_samples*2))
> y_test[0] = 0
> y_test = y_test.to(dev)
> y_test = torch.nonzero(y_test)
> ```
> 
> The above sets `y_test` upper bound to `20` and it's dynamic bound to `19` which is consistent with the characteristics of `x_test`. Wdyt @vanbasten23?

That looks good. I implemented this way and it doesn't fail anymore. As a next step, shall we enable the backward pass and check if it works? Also, since our goal is to run on a mask rcnn model, do you want to create a mask rcnn model and directly test it?

Reply:
Fixed the test.